lib/src/widget/nav_bar/widget/nav_bar_container.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/data/config_pair.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar.dart';
+import 'package:xayn_design/xayn_design.dart';
+
+class NavBarContainer extends StatefulWidget {
+  final MaterialApp child;
+
+  const NavBarContainer({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  @override
+  _NavBarContainerState createState() => _NavBarContainerState();
+
+  /// Call this, when you need to update current instance of the [NavBar]
+  /// f.e. [NavBarConfig] was changed for the screen
+  static void updateNavBar(BuildContext context) =>
```

Comment:
❗ Lets avoid static methods, an idea here is like this:  NavBarContainer.of(context)

Reply:
but `.of(context)` is also static void 😄 
so it is not clear to me what you propose to avoid